sdks/go/pkg/beam/core/metrics/sampler.go

```diff
@@ -117,3 +129,11 @@ func getState(s bundleProcState) string {
 		return ""
 	}
 }
+
+func getRestartLullTimeout() time.Duration {
+	userDefinedTimeout := jobopts.GetElementProcessingTimeout()
```

Comment:
I don't believe Flags are propagated to distributed workers, like for Google Cloud Dataflow. This won't reliably set the value and trigger the value. Consider looking at how the "harnessopts" uses hooks instead.

https://github.com/apache/beam/blob/master/sdks/go/pkg/beam/util/harnessopts/sampler.go

There should be a flag, as you've got it to set the value for users, but it needs to set a pipeline durable mechanism (like option hooks in harnessopts), and access them appropriately.